azure/services/publicips/publicips.go

```diff
@@ -149,3 +149,29 @@ func (s *Service) isIPManaged(ctx context.Context, ipName string) (bool, error)
 	tags := converters.MapToTags(ip.Tags)
 	return tags.HasOwned(s.Scope.ClusterName()), nil
 }
+
+func getPublicIPAddressPropertiesFormat(addressVersion network.IPVersion, dnsSettings *network.PublicIPAddressDNSSettings, outboundIP, outboundIPPrefix net.IP) *network.PublicIPAddressPropertiesFormat {
+	if outboundIP != nil {
```

Comment:
can we set the common fields outside of the conditions and append outbound ip or prefix based on the inputs?

Reply:
I'd also suggest moving the parsing logic inside the helper and parse lazily only if `ip.OutboundIP != nil`

Reply:
something like 
```
pipAddressProperties := &network.PublicIPAddressPropertiesFormat{
			PublicIPAddressVersion:   addressVersion,
			PublicIPAllocationMethod: network.IPAllocationMethodStatic,
			DNSSettings:              dnsSettings,
		}

if ip.outboundip == nil {
return pipAddressProperties
}

// append outbound ip/prefix here

```

Reply:
fixed now